app/components/forms/wizard/sessions-speakers-step.js

```diff
@@ -36,35 +46,56 @@ export default Component.extend(FormMixin, {
   actions: {
     saveDraft() {
       this.onValid(() => {
-        this.get('save')('draft');
+        this.set('data.event.state', 'draft');
+        this.sendAction('save');
       });
     },
     moveForward() {
       this.onValid(() => {
-        this.get('move')();
+        this.sendAction('save');
       });
     },
     publish() {
       this.onValid(() => {
-        this.get('save')('publish');
+        this.set('data.event.state', 'published');
+        this.sendAction('save');
       });
     },
     addItem(type) {
       switch (type) {
         case 'sessionType':
-          this.get('data.sessionsSpeakers.sessionTypes').addObject(this.store.createRecord('session-type'));
+          this.get('data.sessionTypes').addObject(this.store.createRecord('session-type'));
           break;
         case 'track':
-          this.get('data.sessionsSpeakers.tracks').addObject(this.store.createRecord('track'));
+          this.get('data.tracks').addObject(this.store.createRecord('track'));
           break;
         case 'microlocation':
-          this.get('data.sessionsSpeakers.microlocations').addObject(this.store.createRecord('microlocation'));
+          this.get('data.microlocations').addObject(this.store.createRecord('microlocation'));
           break;
       }
     },
-    removeItem(item, type) {
-      item.unloadRecord();
-      this.get(`data.sessionsSpeakers.${type}s`).removeObject(item);
+    removeItem(item) {
+      item.destroyRecord();
```

Comment:
@geekyd There is no point of filtering the arrays by `isDeleted`  if you use `destroyRecord`. `destroyRecord` will instantly make a *DELETE* request to the server  :)